flaml/nlp/autotransformers.py

```diff
@@ -0,0 +1,787 @@
+                if key == "train":
+                    self.train_dataset = merged_folds_from_source.select([x for x in range(target_fold_start, target_fold_end)]).flatten_indices()
+                elif key == "validation":
+                    self.eval_dataset = merged_folds_from_source.select([x for x in range(target_fold_start, target_fold_end)]).flatten_indices()
+                else:
+                    self.test_dataset = merged_folds_from_source.select([x for x in range(target_fold_start, target_fold_end)]).flatten_indices()
+        else:
+            self.train_dataset, self.eval_dataset, self.test_dataset = data_encoded[self._train_name], data_encoded[self._dev_name], data_encoded[
+                self._test_name]
+
+    def _set_model_config(self, checkpoint_path, per_model_config, model_config_num_labels):
+        if per_model_config and len(per_model_config) > 0:
+            model_config = AutoConfig.from_pretrained(
+                checkpoint_path,
+                num_labels=model_config_num_labels,
+                **per_model_config)
+        else:
+            model_config = AutoConfig.from_pretrained(
+                checkpoint_path,
+                num_labels=model_config_num_labels)
+        return model_config
+
+    def  _load_model(self,
+                    checkpoint_path = None,
+                    per_model_config=None):
+        this_task = get_default_task(self.jobid_config.dat[0], self.jobid_config.subdat)
+        if this_task == "seq-classification":
+            self._num_labels = len(self.train_dataset.features["label"].names)
+        elif this_task == "regression":
+            self._num_labels = 1
+
+        if not checkpoint_path:
+            checkpoint_path = self.jobid_config.pre_full
+        if this_task == "seq-classification":
+            num_labels_old = AutoConfig.from_pretrained(checkpoint_path).num_labels
+            if self.jobid_config.pre in MODEL_CLASSIFICATION_HEAD_MAPPING.keys():
+                model_config_num_labels = num_labels_old
+            else:
+                model_config_num_labels = self._num_labels
+            model_config = self._set_model_config(checkpoint_path, per_model_config, model_config_num_labels)
+
+            if self.jobid_config.pre in MODEL_CLASSIFICATION_HEAD_MAPPING.keys():
+                num_labels_old = AutoConfig.from_pretrained(checkpoint_path).num_labels
+                if per_model_config and len(per_model_config) > 0:
+                    model_config = AutoConfig.from_pretrained(
+                        checkpoint_path,
+                        num_labels = num_labels_old,
+                        **per_model_config)
+                else:
+                    model_config = AutoConfig.from_pretrained(
+                        checkpoint_path,
+                        num_labels = num_labels_old)
+
+                if self._num_labels != num_labels_old:
+                    model_config.num_labels = num_labels_old
```

Comment:
unnecessary?

Reply:
Not unecessary. If self._num_labels = 3 and num_labels_old = 2 (converting 2-class into 3 class), model_config.num_labels != num_labels_old before this line

Reply:
the condition `is_pretrained_model_in_classification_head_list()`
is checked in line 308 and `model_config_num_labels = num_labels_old` is already executed